api/placer.py

```diff
@@ -461,47 +463,41 @@ def check(self):
             stamp = minimum
 
         # Remember the timestamp integer for later use with os.utime.
-        self.ziptime = int(dateutil.parser.parse(stamp).strftime('%s'))
+        self.ziptime = dateutil.parser.parse(stamp)
 
         # The zipfile is a santizied acquisition label
         self.dir_ = util.sanitize_string_to_filename(self.a_label)
         self.name = self.dir_ + '.zip'
 
-        # Make a tempdir to store zip until moved
-        # OPPORTUNITY: this is also called in files.py. Could be a util func.
-        self.tempdir = tempfile.TemporaryDirectory(prefix='.tmp', dir=config.get_item('persistent', 'data_path'))
-
         # Create a zip in the tempdir that later gets moved into the CAS.
-        self.path = os.path.join(self.tempdir.name, 'temp.zip')
-        self.zip_  = zipfile.ZipFile(self.path, 'w', zipfile.ZIP_DEFLATED, allowZip64=True)
+        self.path = u'temp.zip'
+        self.zip_ = zipfile.ZipFile(self.file_processor.temp_fs.open(self.path, 'wb'),
+                                    'w', zipfile.ZIP_DEFLATED, allowZip64=True)
 
         # OPPORTUNITY: add zip comment
         # self.zip.comment = json.dumps(metadata, default=metadata_encoder)
 
-        # Bit of a silly hack: write our tempdir directory into the zip (not including its contents).
-        # Creates an empty directory entry in the zip which will hold all the files inside.
-        # This way, when you expand a zip, you'll get folder/things instead of a thousand dicoms splattered everywhere.
-        self.zip_.write(self.tempdir.name, self.dir_)
 
     def process_file_field(self, field, file_attrs):
         # Should not be called with any files
         raise Exception('Files must already be uploaded')
 
     def finalize(self):
 
-        paths = os.listdir(self.folder)
+        paths = self.file_processor.persistent_fs.listdir(self.folder)
         total = len(paths)
 
         # Write all files to zip
         complete = 0
         for path in paths:
-            p = os.path.join(self.folder, path)
```

Comment:
I suggest using `relpath`/`abspath`, or maybe `full_path`, but definitely something different to just `p`.